controllers/provider-alicloud/charts/internal/alicloud-infra/templates/main.tf

```diff
@@ -0,0 +1,115 @@
+provider "alicloud" {
+  access_key = "${var.ACCESS_KEY_ID}"
+  secret_key = "${var.ACCESS_KEY_SECRET}"
+  region = "{{ required "alicloud.region is required" .Values.alicloud.region }}"
+}
+
+// Import an existing public key to build a alicloud key pair
+resource "alicloud_key_pair" "publickey" {
+  key_name = "{{ required "clusterName is required" .Values.clusterName }}-ssh-publickey"
+  public_key = "{{ required "sshPublicKey is required" .Values.sshPublicKey }}"
+}
+
+{{ if .Values.create.vpc -}}
+resource "alicloud_vpc" "vpc" {
+  name       = "{{ required "clusterName is required" .Values.clusterName }}-vpc"
+  cidr_block = "{{ required "vpc.cidr is required" .Values.vpc.cidr }}"
+}
+resource "alicloud_nat_gateway" "nat_gateway" {
+  vpc_id = "{{ required "vpc.id is required" .Values.vpc.id }}"
+  spec   = "Small"
+  name   = "{{ required "clusterName is required" .Values.clusterName }}-natgw"
+}
+{{- end }}
+
+
+// Loop zones
+{{ range $index, $zone := .Values.zones }}
+
+resource "alicloud_vswitch" "vsw_z{{ $index }}" {
+  name              = "{{ required "clusterName is required" $.Values.clusterName }}-{{ required "zone.name is required" $zone.name }}-vsw"
+  vpc_id            = "{{ required "vpc.id is required" $.Values.vpc.id }}"
+  cidr_block        = "{{ required "zone.cidr.worker is required" $zone.cidr.worker }}"
+  availability_zone = "{{ required "zone.name is required" $zone.name }}"
+}
+
+// Create a new EIP.
+resource "alicloud_eip" "eip_natgw_z{{ $index }}" {
+  name                 = "{{ required "clusterName is required" $.Values.clusterName }}-eip-natgw-z{{ $index }}"
+  bandwidth            = "20"
+  internet_charge_type = "PayByBandwidth"
```

Comment:
https://github.com/gardener/gardener/pull/994 is not yet reflected here in this PR - please implement these changes in this extension controller as well